### ISSUE TITLE
Investigate yarn flag `--ignore-optional` behaviour

### ISSUE THREAD

**Issue by hrishikesh-k** [netlify/build-image#699]
Hi!

There seems to be an issue with installing optional dependencies when using Yarn.

The [Netlify documentation](https://docs.netlify.com/configure-builds/manage-dependencies/#yarn) mentions that when using Yarn, `--ignore-optional` flag is passed by default.

While going through the [default build script](https://github.com/netlify/build-image/blob/focal/run-build-functions.sh#L152), this doesn't seem to be the case. I was not able to find a trace of `ignore-optional` in the repo. However, based on [this thread](https://answers.netlify.com/t/yarn-flags-env-variable-being-ignored/33637/23), it appears that optional dependencies are not installed and adding the flag `--no-ignore-optional` doesn't make any difference either.

Could this be investigated? Maybe this was the behaviour in the past, but it changed sometime?

**Comment by JGAntunes** [comment]
@Hrishikesh-K I've hit this myself in the past too. The problem is that the answer is not really in `build-image` but actually coming from `bitballoon`. Background:
- https://github.com/netlify/build-image/issues/128
- https://github.com/netlify/buildbot/issues/268

Seems like this flag was once applied by default but since then we only apply it to a specific yarn version (`0.18.1`). The code:
- https://github.com/netlify/bitballoon/blob/52d3832b3b956ce635fdbc7e1012453530b4f59a/app/models/repo.rb#L33-L36
- https://github.com/netlify/bitballoon/blob/9232e806c4af058d19ed25a4f38a0d0b38a532ea/app/models/build.rb#L798-L801

These env variables are then sent to `buildbot` ☝️

So tl;dr, that option is only added to a really old specific yarn version. I'm curious about the behaviour the customer is reporting, I looked through the thread and didn't see a specific example site (could be useful for us to validate what's happening) because unless they're using yarn `0.18.1` this shouldn't happen afaik 🤔

Also might be worth updating our docs around this flag CC @KyleBlankRollins.

**Comment by JGAntunes** [comment]
@Hrishikesh-K did we get any feedback from the customer? Else, are we good to close this?

**Comment by hrishikesh-k** [comment]
Yeah they never came back. I think we can close this and re-open it later if needed. Thanks.
